go.mod

```diff
@@ -2,6 +2,9 @@ module github.com/terraform-providers/terraform-provider-azurerm
 
 require (
 	github.com/Azure/azure-sdk-for-go v54.3.0+incompatible
+	github.com/Azure/azure-sdk-for-go/sdk/armcore v0.7.1 // indirect
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v0.16.2
+	github.com/Azure/azure-sdk-for-go/sdk/to v0.1.4 // indirect
```

Comment:
As mentioned to the Azure SDK team previously, unfortunately importing these nested Go modules is a hard blocker us - since we do not want to end up importing multiple versions of these in the future when mixing API versions - as such unfortunately we can’t add these dependencies as nested Go modules.

Instead as recommended previously, putting these in a single repository as a single Go module would remove this problem - and makes it easier for consumers to determine when there’s an update to the base library (for example from Github releases).

However as it stands right now unfortunately we can’t import these nested Go modules, so we’ll have to wait for the Azure SDK team to make these changes here.